src/app/components/todo-stats/todo-stats.component.ts

```diff
@@ -1,15 +1,34 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
 
 @Component({
   selector: 'app-todo-stats',
   templateUrl: './todo-stats.component.html',
-  styleUrls: ['./todo-stats.component.scss']
+  styleUrls: ['./todo-stats.component.scss'],
 })
 export class TodoStatsComponent implements OnInit {
+  @Input() taskCount!: number;
+  @Output() getCompleteTaskBtn = new EventEmitter();
+  @Output() getActiveTaskBtn: EventEmitter<boolean> = new EventEmitter();
+  @Output() getAllTaskBtn: EventEmitter<boolean> = new EventEmitter();
+  @Output() clearCompletedBtn: EventEmitter<boolean> = new EventEmitter();
 
-  constructor() { }
+  constructor() {}
 
-  ngOnInit(): void {
+  ngOnInit(): void {}
+
+  showAllTask() {
+    this.getAllTaskBtn.emit();
+  }
+
+  showCompletedTask() {
+    this.getCompleteTaskBtn.emit();
   }
 
+  showActiveTask() {
+    this.getActiveTaskBtn.emit();
+  }
+
+  openModal() {
+    this.clearCompletedBtn.emit();
```

Comment:
You are emit an invalid event in `openModal` function.
you have to implement some code to show the modal component.

Reply:
Here I extract the code that show and hide the modal in the wrapper component .
modal componant that emit amessage for tha parent to take the action of showing it